### PR TITLE
Ease maintaining links for various product flavors

### PR DESCRIPTION
**Pull request by johnjohndoe**
# Description
- Revise configuration of links (event website, source code, Google Play, issues, data privacy statement)  in the _About_ screen (see screenshot) ...
  - to make it less an effort to add a new product flavor.
  - to prevent manual errors when editing the same information over and over.
  - to ease translation of product flavor specific strings.
- Allow to configure custom URLs for _Engelsystem_ JSON export hint in _Settings_ screen (see screenshot) per product flavor.
- Avoid applying customized _c3nav_ URL to all product flavors.

# Affected screens
F1 F2

# Successfully tested on
with `rc3` flavor, `debug` build
- :heavy_check_mark: Pixel 2 device, Android 10

### CODE REVIEW

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/extensions/TextViewExtensions.kt [hunk 1]
```diff
@@ -17,3 +21,19 @@ var TextView.textOrHide: CharSequence
...
+ * Set the given [plainLinkUrl] and the optional [urlTitle] as clickable link to this [TextView].
+ */
+@JvmOverloads
+fun TextView.setLinkText(
+        plainLinkUrl: String,
+        urlTitle: String? = null,
+        movementMethod: MovementMethod,
+        @ColorInt linkTextColor: Int,
+) {
+    val formattedUrl = plainLinkUrl.toLinkified(urlTitle).toSpanned()
```
Comment (cketti) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/extensions/TextViewExtensions.kt:35]: There's no need to create HTML only to convert it to a `Spanned` right away.

```suggestion
    val linkText = urlTitle ?: plainLinkUrl
    val formattedUrl = SpannableString(linkText)
    formattedUrl.setSpan(URLSpan(plainLinkUrl), 0, linkText.length, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
```

Comment (johnjohndoe) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/extensions/TextViewExtensions.kt]: Thank you for the hint. :heavy_check_mark: I simplified it in ca3c00b44466295c77588d8a2e7b3adaeecfa485.

Comment (johnjohndoe) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/extensions/TextViewExtensions.kt]: @cketti Is this good for you? Did you spot any other issues?

**Comment by johnjohndoe** [comment]
_Note:_ The connected Travis CI account has exceeded the limit of free builds. I reached out to the support to upgrade build time.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/extensions/TextViewExtensions.kt (hunk 1 above)
Comment (cketti) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/extensions/TextViewExtensions.kt]: I only gave it a quick look. LGTM

**Pull request merged**